src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -359,6 +361,28 @@ public UUID getFirstSourceDatasetIdFromSnapshotId(
         .orElseThrow(() -> new DatasetNotFoundException("Source dataset for snapshot not found"));
   }
 
+  public SnapshotPreviewModel retrievePreview(UUID snapshotId, String tableName, Integer count) {
+    Snapshot snapshot = retrieve(snapshotId);
+
+    Optional<SnapshotTable> table =
+        snapshot.getTables().stream().filter(t -> t.getName().equals(tableName)).findFirst();
+
+    if (table.isEmpty()) {
+      throw new SnapshotPreviewException("No snapshot table exists with the name: " + tableName);
+    }
```

Comment:
You could write this using `orElseThrow()`, which avoids declaring `table` and the `if`.
```suggestion
    snapshot.getTables().stream().filter(t -> t.getName().equals(tableName))
        .findFirst()
        .orElseThrow(() ->  new SnapshotPreviewException("No snapshot table exists with the name: " + tableName));
```